index.html

```diff
@@ -770,24 +715,41 @@ <h3>
           Receiving inputs
         </h3>
         <p>
-          When the system <dfn>receives new button or axis input values</dfn>,
-          run the following steps:
+          When an [=available=] [=gamepad=] has updated [=input values=], run
+          the following steps [=set/for each=] [=top-level traversable=]
+          |traversable| of the [=user agent=]'s [=user agent/top-level
+          traversable set=]:
         </p>
         <ol>
-          <li>Let |gamepad:Gamepad| be the {{Gamepad}} object representing the
-          device that received new button or axis input values.
-          </li>
-          <li>[=Queue a task=] on the [=gamepad task source=] to [=update
-          gamepad state=] for |gamepad|.
+          <li>[=list/For each=] [=navigable=] of |traversable|'s
+          [=navigable/active document=]'s [=Document/inclusive descendant
+          navigables=]:
+            <ol>
+              <li>Let |window:Window| be |navigable|'s [=navigable/active
+              window=].
+              </li>
+              <li>Let |gamepad:Gamepad| be the {{Gamepad}} in
+              |window|.{{Window/navigator}}.{{Navigator/[[gamepads]]}}
+              representing the [=gamepad=] that has updated [=input values=].
+              </li>
+              <li>[=Queue a global task=] on the [=gamepad task source=] given
+              |window| to [=update gamepad state=] for |gamepad|.
+              </li>
+            </ol>
           </li>
         </ol>
         <p>
           To <dfn>update gamepad state</dfn> for |gamepad:Gamepad|, run the
           following steps:
         </p>
         <ol>
+          <li>If |gamepad|'s [=relevant global object=] is not a {{Window}},
```

Comment:
Maybe add a note here that this might be different on some browsers... we might need a privacy protection here.